sgbatch/src/computation/CopyComputation.cpp

```diff
@@ -0,0 +1,37 @@
+#include <wrench-dev.h>
+
+XBT_LOG_NEW_DEFAULT_CATEGORY(copy_computation, "Log category for CopyComputation");
+
+#include "CopyComputation.h"
+
+CopyComputation::CopyComputation(std::set<std::shared_ptr<wrench::StorageService>> &storage_services,
+                                         std::vector<std::shared_ptr<wrench::DataFile>> &files,
+                                         double total_flops) : CacheComputation::CacheComputation(
+                                             storage_services,
+                                             files,
+                                             total_flops
+                                         ) {}
+
+void CopyComputation::performComputation(std::string &hostname) {
+    WRENCH_INFO("Performing copy computation!");
+    // Incremental size of all input files to process
+    double total_data_size = this->total_data_size;
+    // Read all input files before computation
+    double data_size = 0;
+    for (auto const &fs : this->file_sources) {
+        WRENCH_INFO("Reading file %s from storage service on host %s",
+                    fs.first->getID().c_str(), fs.second->getStorageService()->getHostname().c_str());
+        fs.second->getStorageService()->readFile(fs.first, fs.second);
+        data_size += fs.first->getSize();
+    }
+    //? Question to Henri: How can there be floating point difference?
```

Comment:
Any idea, why these two methods, which should return the same numbers, are different?

Reply:
ah, the magic of floating point comparisons. Anytime in code you're about to write (x == y) where x and y are floats/doubles, you shouldn't. Instead, you'd write something like (std::abs<float>(x - y) < epsilon)).  The order of floating point operations matters, such as additions, subtractions, etc. because float are just an approximation of real numbers... so comparing floats for equality is never a good idea.